app/controllers/ems_cloud_controller.rb

```diff
@@ -235,7 +263,23 @@ def set_ems_record_vars(ems, mode = nil)
 
     ems.azure_tenant_id = params[:azure_tenant_id] if ems.kind_of?(ManageIQ::Providers::Azure::CloudManager)
 
-    ems.update_authentication(build_credentials(ems), :save => (mode != :validate))
+    # ems.update_authentication(build_credentials(ems), :save => (mode != :validate))
+    build_connection(ems, default_endpoint, amqp_endpoint)
+  end
+
+  def build_connection(ems, default_endpoint, amqp_endpoint)
```

Comment:
@durandom, as part of the UI work for pluggable providers, maybe you could start looking at `ems_cloud_controller` for refactoring options ... it feels like the decisions that are made throughout this file are prime candidates for subclassing to provider-specific controllers.